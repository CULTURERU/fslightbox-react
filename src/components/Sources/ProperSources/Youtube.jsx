import React, { Component } from 'react';
import PropTypes from "prop-types";
import { getYoutubeVideoIDFromURL } from "../../../utils/SourceType/getYoutubeVideoIDFromURL";

class Youtube extends Component {
    componentDidMount() {
        this.props.fsLightbox.elements.sources[this.props.i].current.classList.remove('fslightbox-opacity-0');
        if (this.props.fsLightbox.sourcesData.isSourceAlreadyLoadedArray[this.props.i]) {
            return;
        }
        this.props.fsLightbox.sourcesData.sourcesDimensions[this.props.i] = {
            width: 1920,
            height: 1080,
        };
        this.props.onFirstSourceLoad();
    }

    render() {
        return (
            <iframe
                className="fslightbox-single-source fslightbox-opacity-0"
                ref={ this.props.fsLightbox.elements.sources[this.props.i] }
                src={
                    "https://www.youtube.com/embed/"
                    + getYoutubeVideoIDFromURL(this.props.fsLightbox.data.urls[this.props.i])
                    + '?enablejsapi=1'
                }
                allowFullScreen={ true }
                frameBorder="0"
            />
        );
    }
}

Youtube.propTypes = {
    fsLightbox: PropTypes.object.isRequired,
    i: PropTypes.number.isRequired,
    onFirstSourceLoad: PropTypes.func.isRequired,
};
export default Youtube;