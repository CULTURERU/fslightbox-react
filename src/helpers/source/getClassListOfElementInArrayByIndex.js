/**
 * @param { Array } arrayOfElements
 * @param { number } index
 * @return { DOMTokenList }
 */
export const getClassListOfElementInArrayByIndex = (arrayOfElements, index) => {
    return arrayOfElements[index].current.classList;
};