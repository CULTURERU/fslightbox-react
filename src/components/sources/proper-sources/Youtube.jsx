import React, { useEffect } from 'react';
import PropTypes from "prop-types";
import { getYoutubeVideoIdFromUrl } from "../../../helpers/source/getYoutubeVideoIdFromUrl";

/**
 * @param { FsLightbox.data.urls } urls
 * @param { FsLightbox.elements.sources } sources
 * @param { FsLightbox.sourcesData.isSourceAlreadyInitializedArray } isSourceAlreadyInitializedArray
 * @param { FsLightbox.core.sourceController | SetUpSourceController } sourceController
 * @param { number } index
 */
const Youtube = (
    {
        fsLightbox: {
            data: { urls },
            sourcesData: { isSourceAlreadyInitializedArray },
            elements: { sources },
            core: { sourceController }
        },
        index,
    }
) => {
    useEffect(() => {
        sourceController.setIndex(index);
        (isSourceAlreadyInitializedArray[index]) ?
            sourceController.normalLoad() :
            initialLoad();
    });

    const initialLoad = () => {
        sourceController.setSourceWidth(1920);
        sourceController.setSourceHeight(1080);
        sourceController.initialLoad();
    };

    return (
        <iframe
            className="fslightbox-source fslightbox-youtube-iframe fslightbox-opacity-0"
            ref={ sources[index] }
            src={
                "https://www.youtube.com/embed/"
                + getYoutubeVideoIdFromUrl(urls[index])
                + '?enablejsapi=1'
            }
            allowFullScreen={ true }
        />
    );
};

Youtube.propTypes = {
    fsLightbox: PropTypes.object.isRequired,
    index: PropTypes.number.isRequired
};

export default Youtube;