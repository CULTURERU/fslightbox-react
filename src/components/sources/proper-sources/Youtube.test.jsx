import React from 'react';
import { mount } from "enzyme";
import Youtube from "./Youtube";
import * as getYoutubeVideoIdFromUrlObject from "../../../helpers/source/getYoutubeVideoIdFromUrl";

getYoutubeVideoIdFromUrlObject.getYoutubeVideoIdFromUrl = (source) => {
    if (source === 'source') {
        return 'youtube-id';
    } else {
        throw new Error('Invalid param');
    }
};

const fsLightbox = {
    props: { sources: ['source'] },
    elements: { sources: [React.createRef()] },
    collections: { sourcesLoadsHandlers: [{ handleYoutubeLoad: jest.fn() }] }
};

const youtube = mount(<Youtube fsLightbox={ fsLightbox } i={ 0 }/>);

test('useEffect', () => {
    expect(fsLightbox.collections.sourcesLoadsHandlers[0].handleYoutubeLoad).toBeCalled();
});

test('sources ref', () => {
    expect(youtube.getDOMNode()).toBe(fsLightbox.elements.sources[0].current);
});

test('iframe src', () => {
    expect(youtube.getDOMNode().getAttribute('src'))
        .toBe('https://www.youtube.com/embed/youtube-id');
});
