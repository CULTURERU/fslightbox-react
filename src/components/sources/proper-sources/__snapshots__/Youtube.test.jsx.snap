// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<iframe
  allowFullScreen={true}
  className="fslightbox-source fslightbox-opacity-0 fslightbox-youtube-iframe"
  src="https://www.youtube.com/embed/testId?enablejsapi=1"
/>
`;

exports[`DOM 1`] = `
<iframe
  allowFullScreen={true}
  className="fslightbox-source fslightbox-opacity-0 fslightbox-youtube-iframe"
  src="https://www.youtube.com/embed/testId?enablejsapi=1"
/>
`;

exports[`Video DOM static DOM should match snapshot 1`] = `
<iframe
  allowFullScreen={true}
  className="fslightbox-source fslightbox-opacity-0 fslightbox-youtube-iframe"
  src="https://www.youtube.com/embed/testId?enablejsapi=1"
/>
`;
